### DISCUSSION TITLE
What is purpose of a key in filename?

### DISCUSSION THREAD

**Discussion by pedjas** [rustdesk/rustdesk#2119] [General]
When setting up RustDesk client to connect to specific host it is required to add key in file name too. What is exact purpose of that key?

Does this key have to stay the same for all clients connecting to the same host or it has to be different for each client?

**Comment by datenbank-projekt** [comment]
According to https://wiki.articatech.com/network/remote-control/rustdesk/get-the-rustdesk-public-key-or-change-it it's to encrypt communication.

**Reply by pedjas** [reply]
i found it that in meanhile. that is even more confusing. Instead of using standard SSL connection there is soem custom implemented one. And fixed keys are used instead of dynamic. 

Unnecessary complication.

Keys could be generated end exchanged on connection. tha twould provide eve more secure encryption, and no hassle for users to fiddle with fixed keys.
